src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -112,10 +119,24 @@ public TestData(Action action) {
       passCount = totalCount - skipCount - failCount;
 
       failedTests = new ArrayList<String>();
-      failedTestsWithErrorDetail = new ArrayList<FailedTest>();
-      for (TestResult result : testResultAction.getFailedTests()) {
-          failedTests.add(result.getFullName());
-          failedTestsWithErrorDetail.add(new FailedTest(result.getFullName(),result.getErrorDetails()));
+      failedTestsWithErrorDetail = new ArrayList<ExecutedTest>();
+      testListFill((List<TestResult>) testResultAction.getFailedTests(), failedTests, failedTestsWithErrorDetail);
+
+      LogstashConfiguration configuration = LogstashConfiguration.getInstance();
+      if (configuration.isrecordingPassingTests()) {
+          passedTests = new ArrayList<String>();
+          passedTestsWithErrorDetail = new ArrayList<ExecutedTest>();
+          testListFill((List<TestResult>) testResultAction.getPassedTests(), passedTests, passedTestsWithErrorDetail);
+      } else {
+        passedTests = Collections.emptyList();
+        passedTestsWithErrorDetail = Collections.emptyList();
+      }
+    }
+
+    private void testListFill(List<TestResult> testResults, List<String> testNames, List<ExecutedTest> testDetails) {
```

Comment:
Make the signature of the method
`List<? extends TestResult> testResults, List<String> testNames, ...`
Then you don't need the type cast when you call the method